### 1.3. 데이터 확인

|함수|활용방법|
|----|--------|
|df.type()|데이터 형식 파악하기|
|df.head()|처음 n개의 데이터 보이기. 기본 설정은 5|
|df.tail()|마지막 n개의 데이터 보이기. 기본 설정은 5|
|df.shape|몇 행, 몇 열로 구성된 데이터인지 확인|
|df.info()|DataFrame의 총 데이터 수, 컬럼별 데이터 타입, Null 개수|
|df.describe()|DataFrame의 통계 요약 정보. 숫자형 컬럼의 count, mean, std, min, 25%, 50%, 75%, max 값 표시|
|df[].value_counts()|특정 컬럼에 속한 데이터가 각각 몇개 속해 있는지 확인. 예를 들어 표의 등급을 의미하는 Pclass는 1, 2, 3으로 이루어져 있음|
|df.sort_values(by='')|특정 컬럼 값을 기준으로 DataFrame 정렬하기|

1. [Kaggle](https://www.kaggle.com/c/titanic/data)에 가입하기
2. Data Explorer의 train.csv 다운로드 하기
3. 특정 저장 위치(예: "D:/COSADAMA/Week1/train.csv")에 저장하기
4. 2번 예시처럼 저장 위치 복사해 두기
5. pd.read_csv('저장위치/파일이름.csv', encoding = 'utf-8', sep = ',', names = ['원하는','컬럼 이름으로', '변경하기'], usecols = ['원본 데이터의', '원하는', '컬럼이름들'])

In [ ]:
############ 나중에 본인의 경로/train.csv로 바꾸기
import pandas as pd
titanic_df = pd.read_csv('/train.csv')
titanic_df.head()  # 데이터가 너무 많을 때 앞의 5개만 불러오기

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [ ]:
type(titanic_df)

pandas.core.frame.DataFrame

In [ ]:
# 뒤의 5개만 가져오기
titanic_df.tail()

PassengerId  Survived  Pclass                                      Name  \
886          887         0       2                     Montvila, Rev. Juozas   
887          888         1       1              Graham, Miss. Margaret Edith   
888          889         0       3  Johnston, Miss. Catherine Helen "Carrie"   
889          890         1       1                     Behr, Mr. Karl Howell   
890          891         0       3                       Dooley, Mr. Patrick   

        Sex   Age  SibSp  Parch      Ticket   Fare Cabin Embarked  
886    male  27.0      0      0      211536  13.00   NaN        S  
887  female  19.0      0      0      112053  30.00   B42        S  
888  female   NaN      1      2  W./C. 6607  23.45   NaN        S  
889    male  26.0      0      0      111369  30.00  C148        C  
890    male  32.0      0      0      370376   7.75   NaN        Q

In [ ]:
# 앞의 10개만 
titanic_df.head(10)

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   
5            6         0       3   
6            7         0       1   
7            8         0       3   
8            9         1       3   
9           10         1       2   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   
5                                   Moran, Mr. James    male   NaN      0   
6                            McCarthy, Mr. Timothy J    male  54.0      0   
7                     Palsson, Master. Gosta Leonard    male   2.0      3   
8  Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.0      0   
9                Nasser, Mrs. Nicholas (Adele Achem)  female  14.0      1   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S  
5      0            330877   8.4583   NaN        Q  
6      0             17463  51.8625   E46        S  
7      1            349909  21.0750   NaN        S  
8      2            347742  11.1333   NaN        S  
9      0            237736  30.0708   NaN        C

In [ ]:
titanic_df.shape

(891, 12)

In [ ]:
# DataFrame의 총 데이터 건수, 데이터 타입, Null 건수
titanic_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


- 0 to 890: 전체 row 수
- Non-Null Count: 각 컬럼에서 몇개의 데이터가 non-null인지. 전체 891개 중 714개만 존재한다는 건 분명 뭔가 missing data가 있다 의미!
- non-null: null값이 아닌 것.(값이 있다)
- object는 편하게 파이썬에서 배운 문자열이라고 생각하면 됩니다.

In [ ]:
# DataFrame의 통계요약 정보 
titanic_df.describe()

PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200

In [ ]:
# DataFrame 중 'Pclass' 칼럼만 떼어보기 
titanic_Pclass = titanic_df['Pclass']
titanic_Pclass

0      3
1      1
2      3
3      1
4      3
      ..
886    2
887    1
888    3
889    1
890    3
Name: Pclass, Length: 891, dtype: int64

In [ ]:
# 여러 칼럼을 떼어보기 
titanic_df[['Pclass','Age']]   # 여러 개 칼럼은 [ ]로 한번 더 감싸줍니다.

Pclass   Age
0         3  22.0
1         1  38.0
2         3  26.0
3         1  35.0
4         3  35.0
..      ...   ...
886       2  27.0
887       1  19.0
888       3   NaN
889       1  26.0
890       3  32.0

[891 rows x 2 columns]

In [ ]:
# 특정 컬럼의 값이 얼마나 포함되어 있는지 확인
# 컬럼 값의 유형과 건수 확인
# 1등급 216, 2등급 184, 3등급 491석

titanic_df['Pclass'].value_counts()

3    491
1    216
2    184
Name: Pclass, dtype: int64

In [ ]:
# 특정 컬럼 값으로 정렬하기
titanic_df.sort_values(by='Pclass')

PassengerId  Survived  Pclass  \
445          446         1       1   
310          311         1       1   
309          310         1       1   
307          308         1       1   
306          307         1       1   
..           ...       ...     ...   
379          380         0       3   
381          382         1       3   
382          383         0       3   
371          372         0       3   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
445                          Dodge, Master. Washington    male   4.0      0   
310                     Hays, Miss. Margaret Bechstein  female  24.0      0   
309                     Francatelli, Miss. Laura Mabel  female  30.0      0   
307  Penasco y Castellana, Mrs. Victor de Satode (M...  female  17.0      1   
306                            Fleming, Miss. Margaret  female   NaN      0   
..                                                 ...     ...   ...    ...   
379                        Gustafsson, Mr. Karl Gideon    male  19.0      0   
381                        Nakid, Miss. Maria ("Mary")  female   1.0      0   
382                                 Tikkanen, Mr. Juho    male  32.0      0   
371                          Wiklund, Mr. Jakob Alfred    male  18.0      1   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch             Ticket      Fare Cabin Embarked  
445      2              33638   81.8583   A34        S  
310      0              11767   83.1583   C54        C  
309      0           PC 17485   56.9292   E36        C  
307      0           PC 17758  108.9000   C65        C  
306      0              17421  110.8833   NaN        C  
..     ...                ...       ...   ...      ...  
379      0             347069    7.7750   NaN        S  
381      2               2653   15.7417   NaN        C  
382      0  STON/O 2. 3101293    7.9250   NaN        S  
371      0            3101267    6.4958   NaN        S  
890      0             370376    7.7500   NaN        Q  

[891 rows x 12 columns]

In [ ]:
# Pclass와 Name 기준으로 내림차순 정렬하기 
titanic_sorted = titanic_df.sort_values(by=['Pclass','Name'], ascending=False)
titanic_sorted

PassengerId  Survived  Pclass  \
868          869         0       3   
153          154         0       3   
282          283         0       3   
286          287         1       3   
559          560         1       3   
..           ...       ...     ...   
460          461         1       1   
498          499         0       1   
297          298         0       1   
305          306         1       1   
730          731         1       1   

                                                Name     Sex    Age  SibSp  \
868                      van Melkebeke, Mr. Philemon    male    NaN      0   
153                  van Billiard, Mr. Austin Blyler    male  40.50      0   
282                        de Pelsmaeker, Mr. Alfons    male  16.00      0   
286                          de Mulder, Mr. Theodore    male  30.00      0   
559     de Messemaeker, Mrs. Guillaume Joseph (Emma)  female  36.00      1   
..                                               ...     ...    ...    ...   
460                              Anderson, Mr. Harry    male  48.00      0   
498  Allison, Mrs. Hudson J C (Bessie Waldo Daniels)  female  25.00      1   
297                     Allison, Miss. Helen Loraine  female   2.00      1   
305                   Allison, Master. Hudson Trevor    male   0.92      1   
730                    Allen, Miss. Elisabeth Walton  female  29.00      0   

     Parch    Ticket      Fare    Cabin Embarked  
868      0    345777    9.5000      NaN        S  
153      2  A/5. 851   14.5000      NaN        S  
282      0    345778    9.5000      NaN        S  
286      0    345774    9.5000      NaN        S  
559      0    345572   17.4000      NaN        S  
..     ...       ...       ...      ...      ...  
460      0     19952   26.5500      E12        S  
498      2    113781  151.5500  C22 C26        S  
297      2    113781  151.5500  C22 C26        S  
305      2    113781  151.5500  C22 C26        S  
730      0     24160  211.3375       B5        S  

[891 rows x 12 columns]